app/modules/code_provider/code_provider_controller.py

```
@@ -192,3 +192,71 @@ async def check_public_repo(self, repo_name: str) -> bool:
+                    structure.append(
+                        {
+                            "path": item.path,
+                            "type": item.type,  # "blob" for file, "tree" for directory
+                            "size": item.size if item.type == "blob" else None,
+                            "sha": item.sha,
+                        }
+                    )
+
+                return {"structure": structure, "branch": branch_name}
+
+            except Exception as branch_error:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)